crosscluster/physical: TestStreamingZoneConfigsMismatchedRegions failed

crosscluster/physical.TestStreamingZoneConfigsMismatchedRegions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20524653?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20524653?buildTab=artifacts#/) on master @ [cb0be5ffdd1589843614ae5a84114c3d8e726cd2](https://github.com/cockroachdb/cockroach/commits/cb0be5ffdd1589843614ae5a84114c3d8e726cd2):


```
=== RUN   TestStreamingZoneConfigsMismatchedRegions
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/9cca1e0b118933509f3f2f6a2527bece/logTestStreamingZoneConfigsMismatchedRegions2395393345
    test_log_scope.go:82: use -show-logs to present logs inline
    testutils.go:607: 
        	Error Trace:	pkg/crosscluster/replicationtestutils/testutils.go:607
        	            				pkg/crosscluster/replicationtestutils/testutils.go:296
        	            				pkg/crosscluster/physical/replication_stream_e2e_test.go:1513
        	Error:      	Not equal: 
        	            	expected: "replicating"
        	            	actual  : "initial scan on 4 out of 17 ranges"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-replicating
        	            	+initial scan on 4 out of 17 ranges
        	Test:       	TestStreamingZoneConfigsMismatchedRegions
        	Messages:   	job should be in replicating state after reaching start time
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9cca1e0b118933509f3f2f6a2527bece/logTestStreamingZoneConfigsMismatchedRegions2395393345
--- FAIL: TestStreamingZoneConfigsMismatchedRegions (9.15s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStreamingZoneConfigsMismatchedRegions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54820

--- a/pkg/crosscluster/physical/stream_ingestion_frontier_processor.go
+++ b/pkg/crosscluster/physical/stream_ingestion_frontier_processor.go
@@ -86,12 +86,6 @@ type streamIngestionFrontier struct {
 	replicatedTimeAtLastPositiveLagNodeCheck hlc.Timestamp
 
 	rangeStats replicationutils.AggregateRangeStatsCollector
-
-	// This stores the last aggregate stats we computed. Because stats are only
-	// updated on a checkpoint event, the stats will be stale until the next
-	// checkpoint and should not be used to update job statuses. Only on a fresh
-	// checkpoint should we update job statuses.
-	lastAggStats streampb.StreamEvent_RangeStats
 }
 
 var _ execinfra.Processor = &streamIngestionFrontier{}
@@ -345,7 +339,7 @@ func (sf *streamIngestionFrontier) maybeUpdateProgress() error {
 	sf.lastPartitionUpdate = timeutil.Now()
 	log.Dev.VInfof(ctx, 2, "persisting replicated time of %s", replicatedTime)
 
-	statusByStats := sf.aggregateAndUpdateRangeMetrics()
+	sf.aggregateAndUpdateRangeMetrics()
 
 	if err := registry.UpdateJobWithTxn(ctx, jobID, nil /* txn */, func(
 		txn isql.Txn, md jobs.JobMetadata, ju *jobs.JobUpdater,
@@ -361,8 +355,6 @@ func (sf *streamIngestionFrontier) maybeUpdateProgress() error {
 		if replicatedTime.IsSet() && streamProgress.ReplicationStatus == jobspb.InitialScan {
 			streamProgress.ReplicationStatus = jobspb.Replicating
 			md.Progress.StatusMessage = streamProgress.ReplicationStatus.String()
-		} else if statusByStats != "" {
-			md.Progress.StatusMessage = statusByStats
 		}
 
 		// Keep the recorded replicatedTime empty until some advancement has been made
@@ -447,24 +439,13 @@ func (sf *streamIngestionFrontier) maybeCollectRangeStats(
 }
 
 // aggregateAndUpdateRangeMetrics aggregates the range stats collected from each
-// of the ingestion processors and updates the corresponding metrics. If the
-// stats have changed since the last aggregation, it returns a status message
-// to update the job status with. We do this to avoid overwriting job statuses
-// with stale stats as the stats will be the same until the next checkpoint
-// event.
-func (sf *streamIngestionFrontier) aggregateAndUpdateRangeMetrics() string {
-	aggRangeStats, _, statusMsg := sf.rangeStats.RollupStats()
+// of the ingestion processors and updates the corresponding metrics.
+func (sf *streamIngestionFrontier) aggregateAndUpdateRangeMetrics() {
+	aggRangeStats, _, _ := sf.rangeStats.RollupStats()
 	if aggRangeStats.RangeCount != 0 {
 		sf.metrics.ScanningRanges.Update(aggRangeStats.ScanningRangeCount)
 		sf.metrics.CatchupRanges.Update(aggRangeStats.LaggingRangeCount)
 	}
-	if sf.lastAggStats == aggRangeStats {
-		// This is the same stats as last time, so we don't need to update the job
-		// status.
-		return ""
-	}
-	sf.lastAggStats = aggRangeStats
-	return statusMsg
 }
 
 // maybePersistFrontierEntries periodically persists the current state of the

